lib/Parser/Parser.yy

```diff
@@ -50,155 +50,544 @@
+%token <glu::Token> returnKw 19 "return"
+%token <glu::Token> breakKw 20 "break"
+%token <glu::Token> continueKw 21 "continue"
+%token <glu::Token> inKw 22 "in"
+
+%token <glu::Token> trueKw 23 "true"
+%token <glu::Token> falseKw 24 "false"
+%token <glu::Token> asKw 25 "as"
+
+%token <glu::Token> lParen 26 "("
+%token <glu::Token> rParen 27 ")"
+%token <glu::Token> lBrace 28 "{"
+%token <glu::Token> rBrace 29 "}"
+%token <glu::Token> lBracket 30 "["
+%token <glu::Token> rBracket 31 "]"
+%token <glu::Token> dot 32 "."
+%token <glu::Token> comma 33 ","
+%token <glu::Token> colon 34 ":"
+%token <glu::Token> semi 35 ";"
+%token <glu::Token> arrow 36 "->"
+%token <glu::Token> equal 37 "="
+%token <glu::Token> backslash 38 "\\"
+%token <glu::Token> question 39 "?"
+%token <glu::Token> at 40 "@"
+%token <glu::Token> coloncolon 41 "::"
+
+%token <glu::Token> plusOp 42 "+"
+%token <glu::Token> subOp 43 "-"
+%token <glu::Token> mulOp 44 "*"
+%token <glu::Token> divOp 45 "/"
+%token <glu::Token> modOp 46 "%"
+%token <glu::Token> eqOp 47 "=="
+%token <glu::Token> neOp 48 "!="
+%token <glu::Token> ltOp 49 "<"
+%token <glu::Token> leOp 50 "<="
+%token <glu::Token> gtOp 51 ">"
+%token <glu::Token> geOp 52 ">="
+%token <glu::Token> andOp 53 "&&"
+%token <glu::Token> orOp 54 "||"
+%token <glu::Token> bitAndOp 55 "&"
+%token <glu::Token> bitOrOp 56 "|"
+%token <glu::Token> bitXorOp 57 "^"
+%token <glu::Token> bitLShiftOp 58 "<<"
+%token <glu::Token> bitRShiftOp 59 ">>"
+%token <glu::Token> rangeOp 60 "..."
+%token <glu::Token> exclusiveRangeOp 61 "..<"
+%token <glu::Token> notOp 62 "!"
+%token <glu::Token> complOp 63 "~"
+
+%token <glu::Token> intLit 64 "int"
+%token <glu::Token> floatLit 65 "float"
+%token <glu::Token> stringLit 66 "string"
+
+// --- Precedence and associativity declarations ---
+%nonassoc TERNARY
+%nonassoc asPrec
+%left orOp
+%left andOp
+%left eqOp neOp
+%left ltOp leOp gtOp geOp
+%left plusOp subOp
+%left mulOp divOp modOp
+%right notOp complOp
+%right UMINUS
+%nonassoc POSTFIX
+
+%start document
 
 %%
 
-program
-    : /* empty */
-    | declarations
+/*--------------------------------*/
+/* High-level rules               */
+/*--------------------------------*/
+
+document:
+      top_level_list
+    ;
+
+top_level_list:
+      /* empty */
+    | top_level top_level_list
+    ;
+
+top_level:
+      import_declaration
+          { std::cerr << "Parsed top level import declaration" << std::endl; }
+    | type_declaration
+          { std::cerr << "Parsed top level type declaration" << std::endl; }
+    | function_declaration
+          { std::cerr << "Parsed top level function declaration" << std::endl; }
+    ;
+
+attributes:
+      /* empty */
+    | attributes attribute
+    ;
+
+attribute:
+      at ident
+    ;
+
+import_declaration:
+      importKw import_path semi
+    { std::cerr << "Parsed import declaration" << std::endl; }
+    ;
+
+import_path:
+      ident
+    | import_path coloncolon ident
+    | import_path coloncolon mulOp
```

Comment:
here you allow weird stuff:
```
hello::*::world::{a, b::*::stuff}::help::*;
```
this isn't allowed by our formal grammar, and it doesn't make much sense